junit.js

```diff
@@ -32,28 +32,53 @@ function generateJUnitXML(data, options) {
         );
       } else {
         failures++;
-        var failureMessage = ""
-        if (metric.type == "counter") {
-          failureMessage = '"><failure message="failed, count: ' + metric.values.count + '"/></testcase>';
-        }
-        else if (metric.type == "gauge"){
-          failureMessage = '"><failure message="failed, value: ' + metric.values.value + '"/></testcase>';
-        }
-        else if (metric.type == "rate") {
-          failureMessage = '"><failure message="failed, number of fails: ' + metric.values.fails + '"/></testcase>';
-        }
-        else if (metric.type == "trend") {
-          failureMessage = '"><failure message="failed, mean value: ' + metric.values.med + '"/></testcase>';
-        }
-        else {
+        var failureMessage = '';
+        if (metric.type == 'counter') {
+          failureMessage =
+            '"><failure message="failed, count: ' +
+            metric.values.count +
+            ', rate: ' +
+            metric.values.rate +
+            '"/></testcase>';
+        } else if (metric.type == 'gauge') {
+          failureMessage =
+            '"><failure message="failed, value: ' +
+            metric.values.value +
+            ', min: ' +
+            metric.values.min +
+            ', max: ' +
+            metric.values.max +
+            '"/></testcase>';
+        } else if (metric.type == 'rate') {
+          failureMessage =
+            '"><failure message="failed, number of fails: ' +
+            metric.values.fails +
+            ', number of passes: ' +
+            metric.values.passes +
+            ', rate: ' +
+            metric.values.rate +
+            '"/></testcase>';
+        } else if (metric.type == 'trend') {
+          failureMessage =
+            '"><failure message="failed, average value: ' +
+            metric.values.avg +
+            ', min value: ' +
+            metric.values.min +
+            ', mean value: ' +
+            metric.values.med +
+            ', max value: ' +
+            metric.values.max +
+            '"/></testcase>';
```

Comment:
The rest of the PR LGTM, but this part might be problematic. Depending on the value of the [`summaryTrendStats` option](https://k6.io/docs/using-k6/k6-options/reference/#summary-trend-stats), some of these values might not be present in `metric.values`. To be fair, the same thing applies to the `metric.values.med` from the old code, so this was already a problem :disappointed: 

I think we should just programatically build the string here, somewhat like this:
```suggestion
            '"><failure message="failed, ' + 
            Object.entries(metric.values).map(function(kv) {return kv[0] + ' value: ' + kv[1];}).join(', ') +
            '"/></testcase>';
```

In fact, we can probably do this for all metric types, then we won't even need to have the `if (metric.type == 'sth') {}` chain :thinking:

Reply:
I wasn't aware of this option. Fully agree 💯 . Feel free to update the PR. 🙇

Reply:
Updated the PR, please review @na-- / @oleiade 🙇